File: tests/rosetta/transpiler/php/dns-query.php
<?php
ini_set('memory_limit', '-1');
function _lookup_host($host) {
    $res = dns_get_record($host, DNS_A + DNS_AAAA);
    if ($res === false) {
        $fallback = gethostbynamel($host);
        if ($fallback === false) {
            return [[], "lookup failed"];
        }
        return [$fallback, null];
    }
    $ips = [];
    foreach ($res as $r) {
        if (isset($r['ip'])) { $ips[] = $r['ip']; }
        if (isset($r['ipv6'])) { $ips[] = $r['ipv6']; }
    }
    return [$ips, null];
}
$now_seed = 0;
$now_seeded = false;
$s = getenv('MOCHI_NOW_SEED');
if ($s !== false && $s !== '') {
    $now_seed = intval($s);
    $now_seeded = true;
}
function _now() {
    global $now_seed, $now_seeded;
    if ($now_seeded) {
        $now_seed = ($now_seed * 1664525 + 1013904223) % 2147483647;
        return $now_seed;
    }
    return hrtime(true);
}
function _str($x) {
    if (is_array($x)) {
        $isList = array_keys($x) === range(0, count($x) - 1);
        if ($isList) {
            $parts = [];
            foreach ($x as $v) { $parts[] = _str($v); }
            return '[' . implode(' ', $parts) . ']';
        }
        $parts = [];
        foreach ($x as $k => $v) { $parts[] = _str($k) . ':' . _str($v); }
        return 'map[' . implode(' ', $parts) . ']';
    }
    if (is_bool($x)) return $x ? 'true' : 'false';
    if ($x === null) return 'null';
    return strval($x);
}
$__start_mem = memory_get_usage();
$__start = _now();
  $net = ['LookupHost' => '_lookup_host'];
  $res = $net['LookupHost']('www.kame.net');
  $addrs = $res[0];
  $err = $res[1];
  if ($err == null) {
  echo rtrim(_str($addrs)), PHP_EOL;
} else {
  echo rtrim(json_encode($err, 1344)), PHP_EOL;
}
$__end = _now();
$__end_mem = memory_get_usage();
$__duration = max(1, intdiv($__end - $__start, 1000));
$__mem_diff = max(0, $__end_mem - $__start_mem);
$__bench = ["duration_us" => $__duration, "memory_bytes" => $__mem_diff, "name" => "main"];
$__j = json_encode($__bench, 128);
$__j = str_replace("    ", "  ", $__j);
echo $__j, PHP_EOL;

File: tests/rosetta/transpiler/Java/DNS-query.java
public class Main {
    static Object[] res = _netLookupHost("www.kame.net");
    static Object addrs = res[0];
    static Object err = res[1];

    public static void main(String[] args) {
        {
            long _benchStart = _now();
            long _benchMem = _mem();
            if ((err == null)) {
                System.out.println(String.valueOf(addrs));
            } else {
                System.out.println(err);
            }
            long _benchDuration = _now() - _benchStart;
            long _benchMemory = _mem() - _benchMem;
            System.out.println("{");
            System.out.println("  \"duration_us\": " + _benchDuration + ",");
            System.out.println("  \"memory_bytes\": " + _benchMemory + ",");
            System.out.println("  \"name\": \"main\"");
            System.out.println("}");
            return;
        }
    }

    static boolean _nowSeeded = false;
    static int _nowSeed;
    static int _now() {
        if (!_nowSeeded) {
            String s = System.getenv("MOCHI_NOW_SEED");
            if (s != null && !s.isEmpty()) {
                try { _nowSeed = Integer.parseInt(s); _nowSeeded = true; } catch (Exception e) {}
            }
        }
        if (_nowSeeded) {
            _nowSeed = (int)((_nowSeed * 1664525L + 1013904223) % 2147483647);
            return _nowSeed;
        }
        return (int)(System.nanoTime() / 1000);
    }

    static long _mem() {
        Runtime rt = Runtime.getRuntime();
        return rt.totalMemory() - rt.freeMemory();
    }

    static Object[] _netLookupHost(String host) {
        try {
            java.net.InetAddress[] arr = java.net.InetAddress.getAllByName(host);
            String[] out = new String[arr.length];
            for (int i = 0; i < arr.length; i++) { out[i] = arr[i].getHostAddress(); }
            return new Object[]{out, null};
        } catch (Exception e) {
            return new Object[]{null, e.toString()};
        }
    }
}
